contracts/lootbox-v1/contracts/Lootbox.sol

```diff
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./interfaces/IUniswapV2Router02.sol";
+import "../../dollar/contracts/interfaces/IUbiquityAlgorithmicDollarManager.sol";
+import "../../dollar/contracts/interfaces/IUbiquityAlgorithmicDollar.sol";
+import "./interfaces/IWyvernExchange.sol";
+
+contract Lootbox is Ownable {
+
+  IUbiquityAlgorithmicDollarManager public dollarManager;
+  IUbiquityAlgorithmicDollar public dollarToken;
+  IWyvernExchange public wyvernExchange;
+
+  address public creditToken;
+
+  uint256 public transactionFee;
+  uint256 public cashbackPercentage;
+  address private constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+  IUniswapV2Router02 public uniswapRouter;
+
+  modifier exceptBeforeBuy(uint256 _amountDollarToken) {
+
+    string memory dollarTokenSymbol = dollarToken.symbol();
+    string memory requireTextBalance = keccak256(abi.encodePacked("Insufficient Balance of ", dollarTokenSymbol));
+    string memory requireTextAllowance = keccak256(abi.encodePacked("Allowance exceeds of ", dollarTokenSymbol));
+
+    require(IERC20(dollarToken).balanceOf(msg.sender) >= _amountDollarToken * 10**18, requireTextBalance);
+
+    require(IERC20(dollarToken).allowance(msg.sender, address(this)) >= _amountDollarToken * 10**18, requireTextAllowance);
+    _;
+  }
+
+  constructor(
+    IUbiquityAlgorithmicDollarManager _dollarManager,
+    IUniswapV2Router02 _uniswapRouter,
+    IWyvernExchange _wyvernExchange
+  ) {
+    dollarManager = _dollarManager;
+
+    dollarToken = IUbiquityAlgorithmicDollar(dollarManager.dollarTokenAddress());
+    creditToken = dollarManager.debtCouponAddress();
+
+    uCRToken = _uCRToken;
+
+    transactionFee = 25;
+    cashbackPercentage = 100;
```

Comment:
The best practice is 100% => 10,000

```
2.5% => 250
100% => 1000
```

I see the denominator is set as 1,000 so no issue but I think 10,000 is the best practice.

Reply:
I thought best practice is `100^18` but I agree that based on the denominator it should be fine.

Reply:
Okay, then let's deal with 100% as 10,000

Reply:
> The best practice is 100% => 10,000

I agree with @kecrypto :  basis point (BPS) equal to 0.01% is a reference in finance, and is best practice in DeFi also
https://www.investopedia.com/terms/b/basispoint.asp